### ISSUE TITLE
"missing app token" when building with --remote

### ISSUE THREAD

**Issue by coding-in-public** [withastro/astro#10617]
### Astro Info

```block
Astro                    v4.5.1
Node                     v20.11.1
System                   macOS (arm64)
Package Manager          unknown
Output                   static
Adapter                  none
Integrations             astro:db
                         @astrojs/db/file-url
```

### Describe the Bug

I am logged into Astro Studio, linked, and able to work with production data using the --remote flag in dev mode. When attempting to build with the --remote flag, I get the following error (was not happening with astro db 0.8.8, but does happen in 0.9.0): 

Cannot create a remote client: missing app token.
  Stack trace:
    at createRemoteDatabaseClient (file:///Users/chris.pennington/cip/astro-course-files/astro-course-files/node_modules/@astrojs/db/dist/runtime/db-client.js:32:11)
    at ModuleJob.run (node:internal/modules/esm/module_job:218:25)
    at async generatePage (file:///Users/chris.pennington/cip/astro-course-files/astro-course-files/node_modules/astro/dist/core/build/generate.js:204:22)
    at async staticBuild (file:///Users/chris.pennington/cip/astro-course-files/astro-course-files/node_modules/astro/dist/core/build/static-build.js:96:7)
    at async AstroBuilder.run (file:///Users/chris.pennington/cip/astro-course-files/astro-course-files/node_modules/astro/dist/core/build/index.js:165:7)

### What's the expected result?

astro build --remote

should build using the remote data if I’m logged in and linked to an Astro studio project.

(the minimal repo can't really be linked to Astro studio, but short of logging in and linking, everything else is the same as my local files)

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-6icpvd

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by casungo** [comment]
I'm experiencing the exact same issue.

In an attempt to resolve the problem, I tried:
- Deleting the .astro folder
- Reinstalling @astrojs/db
- Reinstalling all packages 
- Checking the documentation to see if the code has changed (my usage of astrodb is very basic, so the code I use is essentially the same as in the docs)

However, none of these actions resolved the issue. 
In the [@astrojs/db changelog](https://github.com/withastro/astro/blob/main/packages/db/CHANGELOG.md#patch-changes-8), the change `Remove embedded app token from CI` is mentioned, and I have a suspicion that this is the culprit modification causing the problem. (Pure speculation)

### Astro info
```
Astro                    v4.5.12
Node                     v20.7.0
System                   Windows (x64)
Package Manager          pnpm
Output                   hybrid
Adapter                  @astrojs/cloudflare
Integrations             @astrojs/sitemap
                         @astrojs/svelte
                         astro:db
                         @astrojs/db/file-url
```

**Comment by matthewp** [comment]
Fixed by https://github.com/withastro/astro/pull/10655

**Comment by rocky8818** [comment]
> Fixed by #10655

In which astro version is the fix?  Im trying in Astro 4.9.2 and astrojs/db 0.11.4
